src/Deployer.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.17;
+
+import {AddressDriver} from "./AddressDriver.sol";
+import {Caller} from "./Caller.sol";
+import {DripsHub} from "./DripsHub.sol";
+import {Managed, ManagedProxy} from "./Managed.sol";
+import {NFTDriver} from "./NFTDriver.sol";
+import {ImmutableSplitsDriver} from "./ImmutableSplitsDriver.sol";
+
+contract Deployer {
+    // slither-disable-next-line immutable-states
+    address public creator;
+
+    DripsHub public dripsHub;
+    bytes public dripsHubArgs;
+    uint32 public dripsHubCycleSecs;
+    DripsHub public dripsHubLogic;
+    bytes public dripsHubLogicArgs;
+    address public dripsHubAdmin;
+
+    Caller public caller;
+    bytes public callerArgs;
+
+    AddressDriver public addressDriver;
+    bytes public addressDriverArgs;
+    AddressDriver public addressDriverLogic;
+    bytes public addressDriverLogicArgs;
+    address public addressDriverAdmin;
+    uint32 public addressDriverId;
+
+    NFTDriver public nftDriver;
+    bytes public nftDriverArgs;
+    NFTDriver public nftDriverLogic;
+    bytes public nftDriverLogicArgs;
+    address public nftDriverAdmin;
+    uint32 public nftDriverId;
+
+    ImmutableSplitsDriver public immutableSplitsDriver;
+    bytes public immutableSplitsDriverArgs;
+    ImmutableSplitsDriver public immutableSplitsDriverLogic;
+    bytes public immutableSplitsDriverLogicArgs;
+    address public immutableSplitsDriverAdmin;
+    uint32 public immutableSplitsDriverId;
+
+    constructor(
+        uint32 dripsHubCycleSecs_,
+        address dripsHubAdmin_,
+        address addressDriverAdmin_,
+        address nftDriverAdmin_,
+        address immutableSplitsDriverAdmin_
+    ) {
+        creator = msg.sender;
+        deployDripsHub(dripsHubCycleSecs_, dripsHubAdmin_);
+        deployCaller();
+        deployAddressDriver(addressDriverAdmin_);
+        deployNFTDriver(nftDriverAdmin_);
+        deployImmutableSplitsDriver(immutableSplitsDriverAdmin_);
+    }
+
+    function deployDripsHub(uint32 dripsHubCycleSecs_, address dripsHubAdmin_) internal {
+        // Deploy logic
+        dripsHubCycleSecs = dripsHubCycleSecs_;
+        dripsHubLogicArgs = abi.encode(dripsHubCycleSecs);
+        dripsHubLogic = new DripsHub(dripsHubCycleSecs);
+        // Deploy proxy
+        dripsHubAdmin = dripsHubAdmin_;
+        // slither-disable-next-line reentrancy-benign
+        ManagedProxy proxy = new ManagedProxy(dripsHubLogic, dripsHubAdmin);
+        dripsHub = DripsHub(address(proxy));
+        dripsHubArgs = abi.encode(dripsHubLogic, dripsHubAdmin);
+    }
+
+    function deployCaller() internal {
+        caller = new Caller();
+        callerArgs = abi.encode();
+    }
+
+    /// @dev Requires DripsHub and Caller to be deployed
+    function deployAddressDriver(address addressDriverAdmin_) internal {
+        // Deploy logic
+        address forwarder = address(caller);
+        uint32 driverId = dripsHub.nextDriverId();
```

Comment:
```suggestion
        uint32 driverId = dripsHub.nextDriverId();
        require(driverId == 0, "incorrect driverId for AddressDriver");
```
It is not needed but I think it is nice to verify some rules

Reply:
We do that in the tests, it makes sure that any changes in the source lead to the right IDs. I don't think that we need that in the code itself, to me it seems noisy and redundant, we're running it on exactly the same smart contracts as tests.

Reply:
Yeah, I saw that. I mean it is not required.